File: Number_Guessing_Games.rb
def guess_the_number
  secret_number = rand(1..100)
  attempts = 0

  puts "Bem-vindo ao jogo de adivinhação!"

  loop do
    print "Digite um número: "
    guess = gets.chomp.to_i
    attempts += 1

    if guess == secret_number
      puts "Parabéns! Você acertou em #{attempts} tentativas."
      break
    elsif guess < secret_number
      puts "Tente um número maior."
    else
      puts "Tente um número menor."
    end
  end
end

guess_the_number

File: Number_Guessing_Games.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	secretNumber := rand.Intn(100) + 1
	attempts := 0

	fmt.Println("Bem-vindo ao jogo de adivinhação!")

	for {
		var guess int
		fmt.Print("Digite um número: ")
		fmt.Scan(&guess)
		attempts++

		if guess == secretNumber {
			fmt.Printf("Parabéns! Você acertou em %d tentativas.\n", attempts)
			break
		} else if guess < secretNumber {
			fmt.Println("Tente um número maior.")
		} else {
			fmt.Println("Tente um número menor.")
		}
	}
}
